### PR TITLE
Rewrite some parts using stream API

### PR DESCRIPTION
**Pull request by PaulKovalov**
I decided to make code a little shorter by using stream API where possible

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -48,24 +51,18 @@
...
-    // provides sorting
-    ArrayList<Event> requiredEvents = new ArrayList<>();
-
     // check if request for a valid duration
     if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
       return Arrays.asList();
     }
 
+    ArrayList<Event> requiredEvents = new ArrayList<>();
+    ArrayList<Event> optionalEvents = new ArrayList<>();
```
Comment (yongli-abc) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:60]: Don't move `optionalEvents` here, it's moved further away from where it's first used.

**Review by yongli-abc — CHANGES_REQUESTED** [review]
IMO I'm conservative about this change. Using stream API doesn't bring any performance gain here, while it makes the code less readable here as the code now becomes a mix of two different coding paradigms (imperative & functional). I know Lambda has its position in many other places (e.g. list comprehension in Python & promise in JS), but IIRC Java doesn't have a heavy smell of functional programming traditionally? Nevertheless I have not used enough Java for a long time :) and perhaps Hemanth can also comment on this.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (PaulKovalov) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:60]: sure, we can just decline this PR, I did this cause personally I like stream api, but im fine with the either way

**Comment by hkveeranki** [comment]
> IMO I'm conservative about this change. Using stream API doesn't bring any performance gain here, while it makes the code less readable here as the code now becomes a mix of two different coding paradigms (imperative & functional). I know Lambda has its position in many other places (e.g. list comprehension in Python & promise in JS), but IIRC Java doesn't have a heavy smell of functional programming traditionally? Nevertheless I have not used enough Java for a long time :) and perhaps Hemanth can also comment on this.

I agree. Functional programming in Java is mostly used during Map Reduce or in similar paradigms. In places like this, I don't think we should be using Lamda. Having said that, I have no idea about performance improvements of stream api. If it isn't, then I wouldn't bother making this change. Infact, I would tend to not make this as the code becomes much less readable.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (yongli-abc) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:60]: Sounds good, but thanks for putting thoughts into exploring a different approach, which is good initiative and I really appreciate that!

**Pull request closed without merging**